expression/builtin_cast.go

```diff
@@ -991,6 +991,10 @@ func (b *builtinCastTimeAsTimeSig) evalTime(row []types.Datum) (res types.Time,
 	if isNull || err != nil {
 		return res, isNull, errors.Trace(err)
 	}
+
+	if res, err = res.Convert(b.tp.Tp); err != nil {
```

Comment:
should we also modify other `builtinCastXXXAsTimeSig` functions ?

Reply:
@zz-jason  Seems no need. No other `cast as time` error found so far.